lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -1252,7 +1252,7 @@ private function executeDeletions(ClassMetadata $class): void
             // Entity with this $oid after deletion treated as NEW, even if the $oid
             // is obtained by a new entity because the old one went out of scope.
             //$this->entityStates[$oid] = self::STATE_NEW;
-            if (! $class->isIdentifierNatural()) {
+            if (! $class->isIdentifierNatural() && !$class->reflFields[$class->identifier[0]]->isReadOnly()) {
```

Comment:
Would it make more sense maybe to modify the readonly relfection object instead and remove that following block and the exception entirely? https://github.com/doctrine/orm/blob/2.13.x/lib/Doctrine/ORM/Mapping/ReflectionReadonlyProperty.php#L45

That way it stays within the reflection abstraction and does not pollute this code here with an implementation detail.

Reply:
That's a way to explore, I stopped it because of 2 questions: 
- Is it OK to ask setValue(), return nothing and don't set the value ? Should we at least return a boolean to tell "false" when the value is not setted?
- What happens in a doctrine world where every Entities in a project have readonly identifiers and therefore aren't nulled when we ask with setValue()? Why do we set ids to null in the first place?